src/main/java/com/laytonsmith/core/events/drivers/InventoryEvents.java

```diff
@@ -987,4 +989,90 @@ public Version since() {
 			return MSVersion.V3_3_1;
 		}
 	}
+
+	@api
+	public static class item_pre_anvil extends AbstractEvent {
+
+		@Override
+		public String getName() {
+			return "item_pre_anvil";
+		}
+
+		@Override
+		public String docs() {
+			return "{}"
+					+ " Fires when a recipe is formed in an anvil, but the result has not yet been clicked."
+					+ " { viewers: all human entities looking at this anvil "
+					+ " | first_item: the first item in the anvil"
+					+ "	| second_item: the second item in the anvil"
+					+ " | result: the result of the anvil"
+					+ " | max_repair_cost: the maximum possible cost of this repair."
+					+ " | level_repair_cost: how many levels are needed to perform this repair?"
+					+ " | item_repair_cost: how many items are needed to perform this repair? }"
+					+ " { result: the result of the anvil }"
+					+ " {}";
+		}
+
+		@Override
+		public boolean matches(Map<String, Mixed> prefilter, BindableEvent event) throws PrefilterNonMatchException {
+			if(event instanceof MCPrepareAnvilEvent) {
+				return true;
+			}
+			return false;
+		}
+
+		@Override
+		public BindableEvent convert(CArray manualObject, Target t) {
+			throw ConfigRuntimeException.CreateUncatchableException("Unsupported operation.", Target.UNKNOWN);
+		}
+
+		@Override
+		public Map<String, Mixed> evaluate(BindableEvent event) throws EventException {
+			if(event instanceof MCPrepareAnvilEvent) {
```

Comment:
Java now has a nice language feature that you can use here.
```java
if(event instanceof MCPrepareAnvilEvent) {
    MCPrepareAnvilEvent e = (MCPrepareAnvilEvent) event;
```
is equivalent to:
```java
if(event instanceof MCPrepareAnvilEvent e) {
```

It's up to you to decide whether or not you want to use that, but it's been used more and more in CH.

Reply:
Sure! I've updated it to use the short-hand.